feat: Loupe search for archived docs

distantnative opened the pull request:
I experienced memory allocation exceeded errors again when I tried to create the index. If we cannot fix this reliably, I am not sure if Loupe will work for us.

distantnative committed:
feat: Loupe search for archived docs

afbora commented:
Getting error for me

<img width="1289" height="392" alt="image" src="https://github.com/user-attachments/assets/ad1037c9-3600-4c13-b89b-f2f92aeda51b" />

distantnative commented:
Pinging @lukasbestle as this is about the security page model.

For me I run again and again in memory allocation errors. It seems very memory hungry, which could be a problem on the server.

lukasbestle commented:
@afbora The error in `SecurityPage::kirbyVersionNoVulnerabilities()` can only occur if `$securityPage->incidents()` is empty. Those are retrieved from the GitHub API. Please try clearing the cache and check if you have the GitHub API key in `site/config/keys.php`.

afbora commented:
Thanks @lukasbestle Yes, after setting GitHub key, I can able to index entire site successfully.

<img width="410" height="76" alt="image" src="https://github.com/user-attachments/assets/15a29acb-fe2e-4d4f-825b-0624ecf63f25" />

distantnative commented:
I realized combining the script and having that called by the GitHub CI doesn't make sense. It works for Algolia cause the index we sent from the GitHub action will be the same as later on the server. But the Loupe database actually needs to end up on the server, not temporarily in the CI action.

afbora commented:
Yes, the Loupe DB will be a static database created only once. We can create it locally and push it. There's no need to run GitHub CI for archived documents.

distantnative committed:
refact: Split search scripts per service

distantnative commented:
I have split the scripts again so that the CI action only runs the Algolia index and the Lupe index can be run manually on the archived version branches to commit the Loupe index database.

I have also added a line to increase the memory limit for the Loupe script as that ran always into errors for me. With the increased memory limit it does work for me.

distantnative committed:
fix: Removed default archived options

distantnative committed:
style: Add cs fixer to search plugin
